symbolic-demangle/src/lib.rs

```diff
@@ -48,38 +50,63 @@ extern "C" {
     fn symbolic_demangle_is_swift_symbol(sym: *const c_char) -> c_int;
 }
 
-/// Defines the output format for demangling.
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
-pub enum DemangleFormat {
-    /// Strips parameter names (in Swift) and sometimes packages or namespaces.
-    Short,
-
-    /// Outputs the full demangled name including arguments, return types, generics and packages.
-    Full,
-}
-
 /// Options for [`Demangle::demangle`].
 ///
+/// The resulting output depends very much on the language of the mangled [`Name`], and not all
+/// options are fully supported by each language.
+///
+/// The default will output a very simple demangling, which means only the function name in most
+/// cases.
+/// Turning on all the options will result in a fully demangled type.
+///
+/// The results on demangling may change over time as more fine grained demangling options and
+/// features are added.
+///
+/// # Examples
+///
+/// ```
+/// use symbolic_common::{Name, Language};
+/// use symbolic_demangle::{Demangle, DemangleOptions};
+///
+/// let symbol = Name::with_language("$s8mangling12curry3ThrowsyyKcyKF", Language::Swift);
+///
+/// let simple = symbol.demangle(DemangleOptions::new()).unwrap();
+/// assert_eq!(&simple, "curry3Throws");
+///
+/// let full = symbol.demangle(DemangleOptions::new().arguments(true).return_type(true)).unwrap();
+/// assert_eq!(&full, "mangling.curry3Throws() throws -> () throws -> ()");
+/// ```
+///
 /// [`Demangle::demangle`]: trait.Demangle.html#tymethod.demangle
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, Default)]
 pub struct DemangleOptions {
-    /// General format to use for the output.
-    ///
-    /// Defaults to `DemangleFormat::Short`.
-    pub format: DemangleFormat,
+    arguments: bool,
+    return_type: bool,
+}
+
+impl DemangleOptions {
+    /// Create new DemangleOptions defaulting to very simple formatting.
+    pub const fn new() -> Self {
+        Self {
+            arguments: false,
+            return_type: false,
```

Comment:
I'm starting to think that we should probably print everything by default, and then have a shorthand to turn off most of the unwanted stuff. A really good example for this is [`DemangleFlags`](https://docs.rs/msvc-demangler/0.8.0/msvc_demangler/struct.DemangleFlags.html): There is a "shorthand" for complete and name only demangling.

What if we provide two base constructors: `DemangleOptions::complete` and `DemangleOptions::name_only`, and they turn everything to `true` and `false`, respectively. Maybe even no `Default` implementation.

FWIW, I'm not a big fan that `NAME_ONLY` is additive to `COMPLETE`, but that's just how `msvc_demangler` works.